src/engine/threaded_engine.h

```diff
@@ -293,6 +299,20 @@ class ThreadedEngine : public Engine {
     finished_cv_.notify_all();
   }
 
+  /*! \brief Return default OMP thread count. Currently, this is whatever OMP shows as number
+   * of procs
+   */
+  static int DefaultOMPThreadsPerWorker() {
+#ifdef _OPENMP
+    // TODO(cjolivier01): Programatically obtain hyperthreading count (if supported)
+    // Taking max including omp_get_max_threads() in case this implementation of OMP accounts for
+    // hyperthreading
+    return std::max(omp_get_max_threads(), omp_get_num_procs());
```

Comment:
why max?

Reply:
It may have been set by the environment variable, or it may have been set elsewhere to something lower.  However, a previous call to set_max... in some library may have reduced it, but we want to either use a larger number in the environment (ie they wish to use hyperthreading number * procs), or the number of procs.
More OMP tuning is in coming PR (including recursion depth, etc)

Reply:
How would user opt to use less number of threads than the number of cores?

Reply:
What's the current use-case for that?

Reply:
Changed to allow environment override.  omp_get_max_threads() can be implementation-specific and may take into account hyperthreading or whatever.  Otherwise, we use # procs (per Eric)